### ISSUE TITLE
Note that become doesn’t work with tty run option.

### ISSUE THREAD

**Issue by arderyp** [deployphp/deployer#2144]
This note is indicated in the documentation here: https://deployer.org/docs/hosts.html

Testing suggests this is correct.

I was wondering if I might be able to find an explanation somewhere of why this is the case?  I am using `become()` and because of this my cloned repo comes out with `myuser:myuser` permissions, which breaks the deployment due to a permissions issue.  I'm wondering if there is some common solution or if I need to just stop using the `become()` approach.

**Comment by antonmedv** [comment]
There is no silver bullet here. One solution is to use one user for server and deployment. Also, ssh into the server using this user. 
Try to avoid unnecessary complexity. Only if you not what you will gain (more security, structure, etc) add something more complex.

**Comment by arderyp** [comment]
Some institutions, my own included, will not allow SSHing into machines through any user but your own.  This is the scenario where `become` is useful.  If you are in this scenario, and you are _also_ forwarding your ssh key, then you seem to be simply out of luck, like me, since this scenario means that you need your agent for git cloning, thus `tty` is required, and yet `become` doesn't work with `tty` enabled.

I've resorted to just deploying as my own user, which works fine.  I just didn't know if there was some technical reason that made it impossible or insecure to use `become` logic with `tty` enabled.  I didn't see anything in your comment @antonmedv explaining this.  I can go ahead an deploy as myself, but it might be worthwhile to document this conflict further and explain why the restriction exists.  Just my $0.02.

**Comment by antonmedv** [comment]
> thus tty is required, and yet become doesn't work with tty enabled.
Yes, this is true and should be documented. 

Did you try to use deploy keys? To avoid using agent forwarding?

**Comment by arderyp** [comment]
as stated in #2145, deploy keys are not an option for me.  Well, they _are_, but the server would need to juggle many keys and the more I looked into it the more that approach really began looking unnecessarily complex--as one of my old managers would put it, "more bees than honey".  Its just much simpler all around in my specific context to deploy as my own user with a forwarded agent, so that's how we've decided to move forward.  Feel free to close if you want.
